lib/lure/repo.go

```diff
@@ -102,12 +105,33 @@ func updateModule(auth Authentication, moduleToUpdate moduleVersion, project Pro
 	} else {
 		dependencyName = moduleToUpdate.Module
 	}
-	title = fmt.Sprintf("Update %s dependency %s to version %s", moduleToUpdate.Type, dependencyName, moduleToUpdate.Latest)
+
+	titlePrefix := fmt.Sprintf("Update %s dependency %s to version ", moduleToUpdate.Type, dependencyName)
+	title = fmt.Sprintf(titlePrefix+"%s", moduleToUpdate.Latest)
+
+	updateVersion, err := ParseVersion(moduleToUpdate.Latest)
+	updateVersionComparable := err == nil
+
 	for _, pr := range existingPRs {
+
 		if pr.Title == title {
 			log.Printf("There already is a PR for: %s", title)
 			return
 		}
+
+		if pr.State == "OPEN" && updateVersionComparable && strings.HasPrefix(pr.Title, titlePrefix) {
+			versionStr := strings.TrimPrefix(pr.Title, titlePrefix)
+			prVersion, err := ParseVersion(versionStr)
+
+			if err == nil && updateVersion.IsGreaterThan(prVersion) {
+				if os.Getenv("DRY_RUN") == "1" {
+					log.Printf("Running in DryRun mode. PR '%s' made for older version '%s' would be declined.", pr.Title, prVersion)
+				} else {
+					log.Printf("Declining PR '%s' made for older version '%s'.", pr.Title, prVersion)
+					declinePullRequest(auth, project.Owner, project.Name, pr.ID)
```

Comment:
Do you think it would be better to decline those after opening the new one? since we get throttled some times, it might decline PRs, but not open a new one. This is a corner case though.

Reply:
Well, I am not sure it is worth it, since I don't see much use in keeping an outdated PR around, even if the new one is not (yet) opened.

Reply:
The only use case I see is when we look at opened PRs to see if there are any libs to bump, but yeah I'm open to let it go.